packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts

```diff
@@ -15,21 +16,63 @@ export const detectDotNetCore = async (): Promise<CheckDependenciesResult> => {
     };
   }
 
-  const result = execa.sync(executableName, ['--version']);
-  const versionString = result.stdout;
+  const sdkResult = execa.sync(executableName, ['--list-sdks']);
+  const installedSdks = sdkResult.stdout;
 
-  if (result.exitCode !== 0) {
-    throw new Error(`${executableName} failed, exit code was ${result.exitCode}`);
+  if (sdkResult.exitCode !== 0) {
+    throw new Error(`${executableName} failed SDK detection, exit code was ${sdkResult.exitCode}`);
   }
+  const sdkInstalled = installedSdks && installedSdks.match(/^3\.1/);
 
-  if (versionString && versionString.startsWith('3.1')) {
+  const toolResult = execa.sync(executableName, ['tool', 'list', '--global']);
+  const installedToolList = toolResult.stdout;
+
+  if (toolResult.exitCode !== 0) {
+    throw new Error(`${executableName} failed tool detection, exit code was ${toolResult.exitCode}`);
+  }
+
+  let toolInstalled = false;
+  let testToolInstalled = false;
+  if (installedToolList) {
+    if (installedToolList.match(/^amazon\.lambda\.tools/m)) {
+      toolInstalled = true;
+    }
+    if (installedToolList.match(/^amazon\.lambda\.testtool-3\.1/m)) {
+      testToolInstalled = true;
+    }
+  }
+
+  if (!toolInstalled) {
+    await installGlobalTool('Amazon.Lambda.Tools');
+  }
+  if (!testToolInstalled) {
+    await installGlobalTool('Amazon.Lambda.TestTool-3.1');
+  }
+
+  // Verify that a dotnet 3.1 SDK and the dotnet Lambda tools is installed locally
+  if (sdkInstalled && toolInstalled) {
```

Comment:
should this condition also include testToolInstalled?

Reply:
Yep! Fixed.